staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go

```diff
@@ -125,6 +126,18 @@ func SerializeObject(mediaType string, encoder runtime.Encoder, innerW http.Resp
 // WriteObjectNegotiated renders an object in the content type negotiated by the client.
 // The context is optional and can be nil.
 func WriteObjectNegotiated(s runtime.NegotiatedSerializer, gv schema.GroupVersion, w http.ResponseWriter, req *http.Request, statusCode int, object runtime.Object) {
+	d, ok := w.(*metrics.ResponseWriterDelegator)
+	if !ok {
+		d = &metrics.ResponseWriterDelegator{ResponseWriter: w}
+		w = d
+	}
+	defer func() {
+		switch {
+		case !d.WroteHeader() && d.ContentLength() == 0:
+			klog.Errorf("WriteObjectNegotiated completed with no write/writeHeader calls:\n%#v\n%#v\n%d\n%#v", s, gv, statusCode, object)
```

Comment:
if this is not a panic case, I'd consider attempting to write a catch-all internal server error here... exiting WriteObjectNegotiated without calling Write or WriteHeader is a bug